### ISSUE TITLE
hub pull-request not using git config correctly

### ISSUE THREAD

**Issue by paulp** [github/hub#832]
```
% git config --get user.email
<my email>
% git pull-request
git config user.email required
```
This happens because user.email is defined via an include; it works if I move it directly into ~/.gitconfig. My ~/.gitconfig contains lines like:
```
[include]
  path = ~/etc/git-config-aliases
  path = ~/etc/git-config-auth
  path = ~/etc/git-config-core
```
This suggests hub is manually dredging the file for config values, which needlessly duplicate's git's logic to the extent that it works and needlessly doesn't work to the extent that it doesn't.

**Comment by paulp** [comment]
I see it's somewhat worse than I had supposed: hub wants to think ~/.gitconfig is a TOML file, and TOML doesn't have includes. This would seem an utterly inexplicable decision in the official github command line client had I not also discovered that the TOM in TOML is the founder of github.

Minimal is a wonderful ambition but it shouldn't be a point of pride when it translates to "doesn't work." .gitconfig isn't a TOML file, it's a git config file. There is a supported interface for obtaining config values from git which is far simpler than introducing a dependency on the wrong configuration language. ```git config --get```.

I apologize if my tone is not well received, which seems likely. But this is by all appearances the worst kind of NIH. If there is some justification for it, I hope you will present it.

**Comment by mislav** [comment]
We switched from TOML a while ago, and we only used it for a brief period for reading `~/.config/hub` and never read `~/.gitconfig` manually. We read git configuration by querying `git config`.

First of all, have you upgraded to latest version of hub?

Second, I don't understand what happens in your original report. Is the creation of pull request aborted with this error message:

```
git config user.email required
```

@jingweno Do you have an idea where this might be coming from?

**Comment by paulp** [comment]
It's the current homebrew hub.
```
% hub version
git version 2.3.1
hub version 2.2.0
```
I tried to figure out where the message was coming from originally once I saw it was likely a dependency, but I failed. I used "Godeps/Godeps.json" as as my list of dependencies - I don't know any go so maybe this is misguided. It was here that I got the idea you were parsing .gitconfig as TOML because you have this dependency
```
		{
			"ImportPath": "github.com/BurntSushi/toml",
			"Comment": "v0.1.0-9-g3883ac1",
			"Rev": "3883ac1ce943878302255f538fce319d23226223"
		},
```
...and since my includes weren't being recognized and that's approximately a git-config parser minus includes, it's pretty strong circumstantial evidence. Sorry for my apparently mistaken conclusion.

To your second question, yes the creation of the pull request was aborted with the error message "git config user.email required", emphasis on was. At the moment I'm not able to reproduce it. Still investigating.

**Comment by jingweno** [comment]
@paulp We wanted to use Toml but it's never enabled - [YAML is used instead](https://github.com/github/hub/blob/master/github/config_service.go#L8-L13) and it's always been YAML. I'm not familiar with your error message. Is it from other programs or git aliases?

**Comment by paulp** [comment]
Found it. Pretty unfortunate. After alias git=hub, "git pull-request" is hub but in the absence of the alias it invokes in my case a program called "git-pull-request" from homebrew's [git-extras](https://github.com/tj/git-extras) formula. That's where the message comes from.

I think I invoked git pull-request in a shell older than the addition of "alias git=hub" thus falling through to the unexpected git-pull-request.

There's a lot wrong with this sequence but none of it is hub's doing. My sincere apologies for the unfounded indignation regarding your only-in-my-head usage of a TOML parser.
